rpc/lib/server/http_server.go

```diff
@@ -31,15 +32,38 @@ const (
 	// same as the net/http default
 	maxHeaderBytes = 1 << 20
 
-	// Timeouts for reading/writing to the http connection.
-	// Public so handlers can read them -
-	// /broadcast_tx_commit has it's own timeout, which should
-	// be less than the WriteTimeout here.
+	// Timeout for reading to the http connection.
+	// Public so handlers can read it.
 	// TODO: use a config instead.
-	ReadTimeout  = 3 * time.Second
-	WriteTimeout = 20 * time.Second
+	ReadTimeout = 3 * time.Second
 )
 
+var (
+	// Timeout for writing to the http connection.
+	// Use the WriteTimeout() function to read this.
+	writeTimeout = 20 * time.Second
+)
+
+// AdjustWriteTimeout adjusts the timeout for writing to the http connection
+// if necessary.
+// That is if and only if the passed in RPCConfig contains
+// a TimeoutBroadcastTxCommit (used by /broadcast_tx_commit) which is larger
+// than the WriteTimeout.
+func AdjustWriteTimeout(cfg config.RPCConfig) (changed bool) {
```

Comment:
let's move this function to node.go

Reply:
`AdjustWriteTimeout` needs to modify the writeTimeout. If we moved the function to node.go, we would need to make the `writeTimeout` a public global var again which isn't `const` (even worse than a global var that only the package controls). Otherwise the function can't modify the var. We shouldn't move the var over to node.go (because we read the value in the rpc package and importing the node package seems the wrong way around).